Network/Public/Include/Network/Handler.hpp

```diff
@@ -33,73 +35,42 @@ class Handler
     virtual ~Handler()            = default;
 
     /// Virtual Handler method for preprocessing next handler
-    virtual Handler*               setNext(Handler* handler)                                                      = 0;
+    virtual std::shared_ptr<Handler> setNext(std::unique_ptr<Handler> handler)                                                      = 0;
```

Comment:
U have a unique as param and u return shared and I guess it's going to be the same Handler or not?

Reply:
@ivanboika  If I return a unique pointer, the code will crash. It has to do with the architecture of this code. In the future, the protected object is used in method calls of this object. I did not change this implementation of the code, so I used a shared pointer. Because when returning a unique pointer, the protected object will be empty. And in the logic of this code, when calling methods, this protected pointer is used and it will be empty if return a unique pointer and the code will crash

Reply:
Have you tried user a shared pointer as a param? Will it solve the issue?

Reply:
Incorrect method definition

Reply:
remove this pseudo-improvement with link return. Keep the pointer in a unique reference. Make a generator function (you can create a separate file) that will combine all the handlers and spawn these handlers in itself